packages/api-fetch/src/index.js

```diff
@@ -166,3 +167,35 @@ apiFetch.fetchAllMiddleware = fetchAllMiddleware;
 apiFetch.mediaUploadMiddleware = mediaUploadMiddleware;
 
 export default apiFetch;
+
+/**
+ * Function that fetches data using apiFetch, and updates the status.
+ *
+ * @param {string} path Query path.
+ */
+export function useApiFetch( path ) {
+	// Indicate the fetching status
+	const [ isLoading, setIsLoading ] = useState( true );
+	const [ data, setData ] = useState( [] );
```

Comment:
I noticed the data is referred to as dataList later on as well. The data returned from an API might also be an object, like when fetching an individual post. I think this could just be kept as an undefined initial value `useState()`.